src/components/ModalPortal.js

```diff
@@ -133,7 +133,7 @@ export default class ModalPortal extends Component {
   }
 
   componentWillUnmount() {
-    if (this.state.isOpen) {
+    if (this.props.isOpen) {
```

Comment:
Perhaps a question for you, @diasbruno, as it's not strictly related to this change -- it looks like the only actual use of `this.state.isOpen` is this block:

```jsx
    // Focus only needs to be set once when the modal is being opened
    if (
      this.props.shouldFocusAfterRender &&
      this.state.isOpen &&
      !prevState.isOpen
    ) {
      this.focusContent();
    }
```

Is that correct? 🤔 Is there a way we could accomplish the programmatic focus of the content without maintaining a `this.state.isOpen` variable alongside `this.props.isOpen`? The simplification may be nice, if so.

Reply:
The state inside the modal is used to manage the timed transition (sync'd with CSS).

When the isOpen flag is changed from true to false, the internal state is true. 
react-modal will wait the css transition N milliseconds, then set to false
(same when you are openning).
When the transition has finished, we then do the tear down any other resources
(applied css names to document.body and others).

I always wanted to implement the transition mechanism using css transition events, 
but never had a change to see the support for it.

Reply:
Ahh! Obrigado pela explicação. :)

Reply:
> I always wanted to implement the transition mechanism using css transition events,
but never had a change to see the support for it.

It would be more reliable than setting timers to manage the internal state.

Reply:
That would be cool! I did see this v4 issue about that: https://github.com/reactjs/react-modal/issues/993

Reply:
This proposed fix does not work if the modal is closed with a timeout. `isOpen` will be `false` when we get here, so `afterClose()` won't run and the body class will be left on the DOM.

Reply:
Are you foregoing support for this?
```TSX
isOpen && (
  <Modal
    isOpen={isOpen}
    closeTimeoutMS={300}
    onRequestClose={() => {
      setIsOpen(false);
    }}
    ...
  />
)
```

If you still want to support the above, I believe `isOpen` will be `false` during unmounting and `afterClose()` won't run leaving the body class in place. Using the `isOpen` props seems unsafe to me, as only the modal component should control when it is open or closed.

Maybe I'm missing something?

Reply:
Wouldn't this solve both problems at once?
```suggestion
    clearTimeout(this.closeTimer);
    if (this.props.isOpen || this.closeTimer) {
      this.closeWithoutTimeout();
    }
```

Reply:
It solves the problem for in my testcase:
```
ModalPortal.js:164 CLOSING MODAL WITH TIMEOUT 300
ModalPortal.js:177 CLOSE WITH TIMEOUT
ModalPortal.js:322 UN-MOUNTING
ModalPortal.js:184 CLOSE WITHOUT TIMEOUT
ModalPortal.js:100 AFTER CLOSE
```